pkg/operatorsource/configuring.go

```diff
@@ -95,20 +137,83 @@ func (r *configuringReconciler) Reconcile(ctx context.Context, in *marketplace.O
 	}
 
 	// If we are here, the given CatalogSourceConfig object already exists.
-	cscNamespacedName := types.NamespacedName{Name: in.Name, Namespace: in.Namespace}
-	cscExisting := marketplace.CatalogSourceConfig{}
-	err = r.client.Get(ctx, cscNamespacedName, &cscExisting)
+	err = r.updateExistingCatalogSourceConfig(ctx, in, packages)
 	if err != nil {
-		r.logger.Errorf("Unexpected error while getting CatalogSourceConfig: %s", err.Error())
+		r.logger.Errorf("Unexpected error while updating CatalogSourceConfig: %s", err.Error())
 		nextPhase = phase.GetNextWithMessage(phase.Configuring, err.Error())
-
 		return
 	}
 
+	r.logger.Info("CatalogSourceConfig object has been updated successfully")
+
+	nextPhase = phase.GetNext(phase.Succeeded)
+	return
+}
+
+// getManifestMetadata gets the package metadata from the OperatorSource endpoint.
+// It returns the list of packages to be written to the OperatorSource status. error is set
+// when there is an issue downloading the metadata. In that case the list of packages
+// will be empty.
+func (r *configuringReconciler) getManifestMetadata(spec *marketplace.OperatorSourceSpec, namespace string) ([]*datastore.RegistryMetadata, error) {
+
+	metadata := make([]*datastore.RegistryMetadata, 0)
+
+	options, err := SetupAppRegistryOptions(r.client, spec, namespace)
+	if err != nil {
+		return metadata, err
+	}
+
+	registry, err := r.factory.New(options)
+	if err != nil {
+		return metadata, err
+	}
+
+	metadata, err = registry.ListPackages(spec.RegistryNamespace)
+	if err != nil {
+		return metadata, err
+	}
+
+	return metadata, nil
+}
+
+// writeDatastoreManifests checks to see if there are any manifests
```

Comment:
Again, we are not writing manifests to the datastore but just the metadata. Please fix all instances of the places where you are using `manifests`.

Reply:
does this include renaming the function to `writeDatastoreMetadata`?

Reply:
Yes. Suggestion: `writeMetadataToDatastore`